nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscriptions(parentDocModel);
+
+        // create a child document
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
+
+        // fire the dummy event on the child document
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), mainDocModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(mainDocModel, ANOTHER_DOCUMENT_NAME);
+    }
+
+    protected void checkMailContent(final DocumentModel documentModel, String documentName) {
+        assertEquals(1, emailsResult.getMails().size());
+        SmtpMailServerFeature.MailMessage mailMessage = emailsResult.getMails().get(0);
+        assertEquals(String.format("[Dummy]Notification on the document '%s'", documentName), mailMessage.getSubject());
+
+        // check the mail content (see templates/dummyNotificationToSendMail.ftl)
+        String mailContent = getMailContent(mailMessage);
+        // description
+        assertTrue(mailContent.contains(String.format("The document %s is now available.", documentName)));
+        // author
+        assertTrue(mailContent.contains(session.getPrincipal().getName()));
+        // location
+        assertTrue(mailContent.contains(documentModel.getPathAsString()));
+        // version
+        assertTrue(mailContent.contains(documentModel.getVersionLabel()));
+        // state
+        assertTrue(mailContent.contains(documentModel.getCurrentLifeCycleState()));
+        // link to the document
+        assertTrue(mailContent.contains(String.format("Consult the document %s", documentName)));
+    }
+
+    protected DocumentModel createDocument(DocumentModel parent, String name) {
+        DocumentModel documentModel = session.createDocumentModel(parent.getPathAsString(), name, "File");
+        return session.createDocument(documentModel);
+    }
+
+    protected void addSubscriptions(DocumentModel documentModel) {
+        NuxeoPrincipal principal = session.getPrincipal();
+        String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
+        notificationManager.addSubscription(subscriber, DUMMY_NOTIFICATION_NAME, documentModel, false, principal,
+                DUMMY_NOTIFICATION_NAME);
+    }
+
+    protected String getMailContent(SmtpMailServerFeature.MailMessage mailMessage) {
+        String content = mailMessage.getContent();
+        if (!mailMessage.getContentType().contains("text/html")) {
+            return content;
```

Comment:
Why this check, can't you assert that we get HTML? There's no variability here, is there?

Reply:
my idea is to be closer to the real use case where we have a html content and the idea is just to extract the text and verify it to the expected text